src/wp-admin/includes/image.php

```diff
@@ -914,9 +913,52 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
 					// Save the meta data before any image post-processing errors could happen.
 					wp_update_attachment_metadata( $attachment_id, $metadata );
 
-					// Create sub-sizes saving the image meta after each.
-					$metadata = _wp_make_subsizes( $merged_sizes, $image_file, $metadata, $attachment_id, '' );
+					$primary_mime_type = isset( $metadata['sizes']['full']['mime-type'] ) ? $metadata['sizes']['full']['mime-type'] : 'image/jpeg';
+
+					// Get mime types to generate.
+					$transforms      = wp_upload_image_mime_transforms( $attachment_id );
+					$mime_transforms = array();
+
+					if ( ! empty( $transforms[ $primary_mime_type ] ) ) {
+						$mime_transforms = $transforms[ $primary_mime_type ];
+					}
+
+					// Generate image sub-sizes for each mime type.
+					if ( ! empty( $mime_transforms ) ) {
+						foreach ( $mime_transforms as $mime_type ) {
```

Comment:
generate the images for the primary type first, then loop thru the secondary types